zebra-state/src/in_memory/block_index.rs

```diff
@@ -22,45 +22,28 @@ impl BlockIndex {
         let hash = block.as_ref().into();
         let height = block.coinbase_height().unwrap();
 
-        match self.by_height.entry(height) {
+        match self.height_map.entry(height) {
             Entry::Vacant(entry) => {
-                let _ = entry.insert(block.clone());
+                let _ = entry.insert(hash);
                 let _ = self.by_hash.insert(hash, block);
                 Ok(hash)
             }
             Entry::Occupied(_) => Err("forks in the chain aren't supported yet")?,
         }
     }
 
-    pub(super) fn get(&mut self, query: impl Into<BlockQuery>) -> Option<Arc<Block>> {
-        match query.into() {
-            BlockQuery::ByHash(hash) => self.by_hash.get(&hash),
-            BlockQuery::ByHeight(height) => self.by_height.get(&height),
-        }
-        .cloned()
+    pub(super) fn get(&self, hash: BlockHeaderHash) -> Option<Arc<Block>> {
+        self.by_hash.get(&hash).cloned()
     }
 
-    pub(super) fn get_tip(&self) -> Option<Arc<Block>> {
-        self.by_height
-            .iter()
-            .next_back()
-            .map(|(_key, value)| value.clone())
+    pub(super) fn get_at(&self, height: BlockHeight) -> Option<BlockHeaderHash> {
+        self.height_map.get(&height).cloned()
```

Comment:
There's a significant difference between get(hash) and get(height):
* hash guarantees that it will return the exact block, if it exists
* height uses the current main chain

That's not documented, and I don't want to block this PR on function documentation.

But let's make the difference more explicit in the function names:
```suggestion
    pub(super) fn get_main_chain_at(&self, height: BlockHeight) -> Option<BlockHeaderHash> {
        self.height_map.get(&height).cloned()
```

Reply:
Fair enough, for now it doesn't matter because the on_disk will still only support 1 chain but eventually that is likely to change.